src/javascript/app_2/App/Middlewares/prevent_blacklisted_landing_companies.js

```diff
@@ -0,0 +1,3 @@
+import Client                      from '_common/base/client_base';
+
+export default () => Client.hasCostaricaAccount() || (Client.getAccountType() === 'virtual');
```

Comment:
it's better to name the function.

Reply:
@binary-hamedanchi What kind of race condition?

Reply:
@easteregg Whether to receive the `authorize` response before calling this function or not.

Reply:
Checked this with @negar-binary, couldn't find a situation where there is any problem regarding showing/flashing this dialog to the wrong user/account. Mainly because way before this check is going to perform, we already know the client current loginid.

Reply:
@binary-hamedanchi yes, even on login to the page, we already have called authorize on logged_in page and keep the is_virtual value in storage, so there's no problem with detecting virtual accounts. unless we missed some scenario that you are thinking of?